### PR TITLE
Save the Continuous b-tagging quantiles, SF, & ineffSFs to Ntuple

### PR DESCRIPTION
**Pull request by nhartman94**

### CODE REVIEW

File: Root/BJetEfficiencyCorrector.cxx [hunk 1]
```diff
@@ -130,7 +133,9 @@ EL::StatusCode BJetEfficiencyCorrector :: initialize ()
   if (m_taggerName == "DL1rmu") { taggerOK = true; m_getScaleFactors =  false; }
 
   // Continuous
-  if (m_operatingPt == "Continuous") { opOK = true; taggerOK = true; m_getScaleFactors =  true; m_useContinuous = true; ANA_MSG_DEBUG(" Using continuous b-tagging");}
+  // if (m_operatingPt == "Continuous") { opOK = true; taggerOK = true; m_getScaleFactors =  true; m_useContinuous = true; ANA_MSG_DEBUG(" Using continuous b-tagging");}
+  //m_useContinuous = static_cast<bool>(m_operatingPt == "Continuous");
+  //if (m_useContinuous) ANA_MSG_DEBUG(" Using continuous b-tagging");
```
Comment (jbossios) [inline Root/BJetEfficiencyCorrector.cxx]: Could you please remove the commented code?

Comment (nhartman94) [reply Root/BJetEfficiencyCorrector.cxx]: Hi yes, ofc, one second.

**Comment by nhartman94** [comment]
> Why are there so many changes in BJetEfficiencyCorrector when my understanding is that the only missing thing was to save what we already have into a TTree.

It looked like there was originally some logic where it was only looking for the SF if dec_isBTag() wasTrue, but this decoration never got in the continuous mode since we use the dec_quantile() instead. I tried to just shuffle a few lines to resolve this.

**Comment by nhartman94** [comment]
> > Why are there so many changes in BJetEfficiencyCorrector when my understanding is that the only missing thing was to save what we already have into a TTree.
> 
> It looked like there was originally some logic where it was only looking for the SF if dec_isBTag() wasTrue, but this decoration never got in the continuous mode since we use the dec_quantile() instead. I tried to just shuffle a few lines to resolve this.

I was following your discussion in https://github.com/UCATLAS/xAODAnaHelpers/issues/1372 for starting these edits.

**Comment by nhartman94** [comment]
Also tagging @mswiatlo and @rateixei since we want to use this for the HH->4b analysis

**Comment by nhartman94** [comment]
> > Why are there so many changes in BJetEfficiencyCorrector when my understanding is that the only missing thing was to save what we already have into a TTree.
> 
> It looked like there was originally some logic where it was only looking for the SF if dec_isBTag() wasTrue, but this decoration never got in the continuous mode since we use the dec_quantile() instead. I tried to just shuffle a few lines to resolve this.

As in, this line was never true when I was running in the Continuous mode

**Comment by jbossios** [comment]
You are right the filling of SF/InefficiencySFs was not right in the continuous case. Thanks for the fix!

**Comment by kkrizka** [comment]
Can you remove changes that are only changing the tabulation?

Otherwise I think this looks good.

**Comment by nhartman94** [comment]
> Can you remove changes that are only changing the tabulation?
> 
> Otherwise I think this looks good.

Ok, thank you for the comments. I'm working on undo-ing my indentation edits, so it'll take a few more commits to make sure I'm doing it all correctly, but I will comment again when I've completed this.

**Comment by nhartman94** [comment]
Hi @kkrizka - I finished taking out the indentation edits, so hopefully it's easier for you to review my edits now. Please let me know if you need anything else from my side.

**Comment by rateixei** [comment]
Hi, @jbossios and @kkrizka - is there anything missing to this MR for it to be merged? Thanks!

**Comment by kkrizka** [comment]
Looks good to me, I think.

**Pull request merged**